flink-kubernetes-webhook/src/test/java/org/apache/flink/kubernetes/operator/admission/AdmissionHandlerTest.java

```diff
@@ -0,0 +1,104 @@
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/** @link AdmissionHandler unit tests */
+public class AdmissionHandlerTest {
+
+    private final AdmissionHandler admissionHandler =
+            new AdmissionHandler(new FlinkValidator(new DefaultDeploymentValidator()));
+
+    @Test
+    public void testHandleIllegalRequest() {
+        final EmbeddedChannel embeddedChannel = new EmbeddedChannel(admissionHandler);
+        final String illegalRequest = "/test";
+        embeddedChannel.writeInbound(new DefaultFullHttpRequest(HTTP_1_1, GET, illegalRequest));
+        embeddedChannel.writeOutbound(new DefaultFullHttpResponse(HTTP_1_1, OK));
+        final DefaultFullHttpResponse response = embeddedChannel.readOutbound();
+        assertEquals(INTERNAL_SERVER_ERROR, response.status());
+        assertEquals(
+                String.format(
+                        "Illegal path requested: %s. Only %s is accepted.",
+                        illegalRequest, VALIDATE_REQUEST_PATH),
+                new String(response.content().array()));
+        assertTrue(embeddedChannel.finish());
+    }
+
+    @Test
+    public void testHandleValidateRequestWithoutContent() {
+        final EmbeddedChannel embeddedChannel = new EmbeddedChannel(admissionHandler);
+        embeddedChannel.writeInbound(
+                new DefaultFullHttpRequest(HTTP_1_1, GET, VALIDATE_REQUEST_PATH));
+        embeddedChannel.writeOutbound(new DefaultFullHttpResponse(HTTP_1_1, OK));
+        final DefaultFullHttpResponse response = embeddedChannel.readOutbound();
+        assertEquals(INTERNAL_SERVER_ERROR, response.status());
+        assertTrue(
+                new String(response.content().array())
+                        .contains(MismatchedInputException.class.getName()));
+        assertTrue(embeddedChannel.finish());
+    }
+
+    @Test
+    public void testHandleValidateRequestWithAdmissionReview() throws IOException {
+        final EmbeddedChannel embeddedChannel = new EmbeddedChannel(admissionHandler);
+        embeddedChannel.writeInbound(
+                new DefaultFullHttpRequest(
+                        HTTP_1_1,
+                        GET,
+                        VALIDATE_REQUEST_PATH,
+                        Unpooled.wrappedBuffer(
+                                FileUtils.readFileUtf8(
+                                                new File(
+                                                        Objects.requireNonNull(
+                                                                        getClass()
+                                                                                .getClassLoader()
+                                                                                .getResource(
+                                                                                        "admission-review.json"))
+                                                                .getPath()))
```

Comment:
Could we simply use the objectmapper to generate the request instead of hardcoding the json?

Reply:
@gyfora, of course. IMO. the current implementation doesn't look so good to me.